yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -368,69 +375,70 @@ class SkipOperator : public ObservableOperator<T, T> {
-class IgnoreElementsOperator : public ObservableOperator<T, T> {
+class IgnoreElementsOperator
+    : public ObservableOperator<T, T, IgnoreElementsOperator<T>> {
+  using ThisOperatorT = IgnoreElementsOperator<T>;
+  using Super = ObservableOperator<T, T, ThisOperatorT>;
+
  public:
   explicit IgnoreElementsOperator(Reference<Observable<T>> upstream)
-      : ObservableOperator<T, T>(std::move(upstream)) {}
+      : Super(std::move(upstream)) {}
 
   void subscribe(Reference<Observer<T>> observer) override {
-    ObservableOperator<T, T>::upstream_->subscribe(
+    Super::upstream_->subscribe(
         make_ref<Subscription>(get_ref(this), std::move(observer)));
   }
 
  private:
-  class Subscription : public ObservableOperator<T, T>::Subscription {
-    using Super = typename ObservableOperator<T, T>::Subscription;
-
+  class Subscription : public Super::Subscription {
    public:
     Subscription(
-        Reference<Observable<T>> observable,
+        Reference<ThisOperatorT> observable,
         Reference<Observer<T>> observer)
-        : ObservableOperator<T, T>::Subscription(
-              std::move(observable),
-              std::move(observer)) {}
+        : Super::Subscription(std::move(observable), std::move(observer)) {}
 
     void onNext(T) override {}
   };
 };
 
 template <typename T>
-class SubscribeOnOperator : public ObservableOperator<T, T> {
+class SubscribeOnOperator
+    : public ObservableOperator<T, T, SubscribeOnOperator<T>> {
+  using ThisOperatorT = SubscribeOnOperator<T>;
+  using Super = ObservableOperator<T, T, ThisOperatorT>;
+
  public:
   SubscribeOnOperator(Reference<Observable<T>> upstream, Scheduler& scheduler)
-      : ObservableOperator<T, T>(std::move(upstream)),
-        worker_(scheduler.createWorker()) {}
+      : Super(std::move(upstream)), worker_(scheduler.createWorker()) {}
 
   void subscribe(Reference<Observer<T>> observer) override {
-    ObservableOperator<T, T>::upstream_->subscribe(make_ref<Subscription>(
+    Super::upstream_->subscribe(make_ref<Subscription>(
         get_ref(this), std::move(worker_), std::move(observer)));
   }
 
  private:
-  class Subscription : public ObservableOperator<T, T>::Subscription {
+  class Subscription : public Super::Subscription {
    public:
     Subscription(
-        Reference<Observable<T>> observable,
+        Reference<ThisOperatorT> observable,
         std::unique_ptr<Worker> worker,
         Reference<Observer<T>> observer)
-        : ObservableOperator<T, T>::Subscription(
-              std::move(observable),
-              std::move(observer)),
+        : Super::Subscription(std::move(observable), std::move(observer)),
           worker_(std::move(worker)) {}
 
     void cancel() override {
       worker_->schedule([this] { this->callSuperCancel(); });
     }
 
     void onNext(T value) override {
-      auto* observer = ObservableOperator<T, T>::Subscription::observer_.get();
+      auto observer = Super::Subscription::observer_;
```

Comment:
Can we use `using` keyword instead of this line? 
```
 class Subscription : public Super::Subscription {
     public:
        using Super::Subscription::observer_;
     ...
```

Reply:
We could, I'm not convinced this makes it more apparent what the code is doing or where `observer_` comes from though